website/pages/docs/k8s/upgrade/index.mdx

```diff
@@ -7,63 +7,303 @@ description: Upgrade Consul on Kubernetes
+```
+
+**Before upgrading, be sure you've read the other sections on this page.**
+
+### Consul Version Upgrade
+
+If a new version of Consul is released, you will need to run a Helm upgrade
+to update to the new version.
+
+1. Ensure you've read the [Upgrading Consul](/docs/upgrading) documentation.
+1. Ensure you've read any [specific instructions](/docs/upgrading/upgrade-specific) for the version you're upgrading
+   to and the Consul [changelog](https://github.com/hashicorp/consul/blob/master/CHANGELOG.md) for that version.
+1. Read our [Compatibility Matrix](/docs/k8s/upgrade/compatibility) to ensure
+   your current Helm chart version supports this Consul version. If it does not,
+   you may need to also upgrade your Helm chart version at the same time.
+1. Set `global.consul` in your `values.yaml` to the desired version:
+   ```yaml
+   global:
+     image: consul:1.8.3
+   ```
+1. Determine your current installed chart version
+
+```bash
+helm list -f consul
+NAME  	NAMESPACE	REVISION	UPDATED        	STATUS  	CHART        	APP VERSION
+consul	default  	2       	2020-09-30 ...	deployed	consul-0.24.0	1.8.2
+```
+
+In this example, I'm running version `0.24.0` (from `consul-0.24.0`).
+
+1. Run `helm upgrade consul hashicorp/consul --version 0.24.0 -f /path/to/my/values.yaml`.
+
+   **Before upgrading, be sure you've read the other sections on this page.**
+
+~> NOTE: It's important to always set the `--version` flag because otherwise Helm
+will use the most up-to-date version in its local cache which may result in an
+unintended upgrade.
+
+## Determining What Will Change
+
+Before upgrading, it's important to understand what changes will be made to your
+cluster. For example, you will need to take more care if your upgrade will result
+in the Consul server statefulset being redeployed.
+
+There is no built-in functionality to Helm that shows what a helm upgrade will
+change. There is however a Helm plugin [helm-diff](https://github.com/databus23/helm-diff)
+that can be used.
+
+1. Install `helm-diff` with:
+
+```bash
+helm plugin install https://github.com/databus23/helm-diff
+```
+
+1. If you are updating your `values.yaml` file, do so now.
+1. Take the same `helm upgrade` command you were planning to execute but run `helm diff upgrade` instead of `helm upgrade`:
+
+```bash
+helm diff upgrade consul hashicorp/consul --version 0.24.1 -f /path/to/your/values.yaml
+```
+
+This will print out the manifests that will be updated and their diffs.
+
+1. To see only the objects that will be updated, add `| grep "has changed"`:
+
+```bash
+helm diff upgrade consul hashicorp/consul --version 0.24.1 -f /path/to/your/values.yaml |
+  grep "has changed"
+```
+
+1. Take specific note if `consul, DaemonSet` or `consul-server, StatefulSet` are listed.
+   This means that your Consul client daemonset or Consul server statefulset (or both) will be redeployed.
+
+If either is being redeployed, we will follow the same pattern for upgrades as
```

Comment:
```suggestion
If either is being redeployed, the consul-k8s binary will follow the same pattern for upgrades as
```

Reply:
Yeah this is saying "if the daemonset or server is being redeployed then the person doing the upgrade will need to follow the same pattern as other platforms"